### ISSUE TITLE
Unable to write pin to RSC 

### ISSUE THREAD

**Issue by mxblsdl** [rstudio/pins-python#148]
When trying to write any dataset to RSC with python pins I get an error related to a temporary file that does not exist. In the code below I am user L170489.
```
import pins
from pins.data import mtcars
from dotenv import load_dotenv

load_dotenv() # load the api key

board = pins.board_rsconnect()

board.pin_write(mtcars, "L170489/mtcars", type="csv")
```
Results in the following error.
```
Writing pin:
Name: 'L170489/mtcars'
Version: 20220720T101415Z-4df76
Traceback (most recent call last):
  File "c:/local/r_projects/command_center/py/test.py", line 9, in <module>
    board.pin_write(mtcars, "L170489/mtcars", type="csv")
  File "C:\Users\L170489\AppData\Local\Programs\Python\Python37\lib\site-packages\pins\boards.py", line 784, in pin_write
    meta = f_super(*args, **kwargs)
  File "C:\Users\L170489\AppData\Local\Programs\Python\Python37\lib\site-packages\pins\boards.py", line 313, in pin_write
    res = self.fs.put(tmp_dir, dst_version_path, recursive=True)
  File "C:\Users\L170489\AppData\Local\Programs\Python\Python37\lib\site-packages\pins\rsconnect\fs.py", line 214, in put
    bundle = self.api.post_content_bundle(content["guid"], lpath)
  File "C:\Users\L170489\AppData\Local\Programs\Python\Python37\lib\site-packages\pins\rsconnect\api.py", line 382, in post_content_bundle
    with open(tmp.name, "rb") as f:
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\L170489\\AppData\\Local\\Temp\\tmp88zehwr5.tar.gz'
```
On RSC the dataset will partially show up with the error:
![image](https://user-images.githubusercontent.com/37624866/180043648-8dab81a9-c933-470f-8e98-20bf64a51e14.png)

I am unable to find any information on this error and do not understand why it is showing up. I am on Windows 10.

**Comment by machow** [comment]
Thanks for raising--it looks like maybe a cross-platform path issue in pins. I've opened a PR that adds windows to our CI tests, and will add a test to make sure the path here works okay on windows...

**Comment by heath-r** [comment]
A user with Mondelēz International has also reported this issue in Windows to the RStudio Support Team. Is there an update about applying a fix for Windows users?

**Comment by machow** [comment]
No update on timeline--but I think the issue is with this piece of code:

https://github.com/rstudio/pins-python/blob/main/pins/rsconnect/api.py#L374-L380

Our use of `.close()` looks like it's deleting the temporary file on windows. We should use a temporary directory instead of temporary file, or use some combo of `.flush()` and `.sync()` instead?

**Comment by trentrichardson** [comment]
Just wanted to mention I got this same error on Windows 10. I moved my notebook to a linux development environment and it ran fine.

**Comment by machow** [comment]
I've released pins v0.0.6, which runs tests on windows machine, and added a potential fix for this issue (the way we were closing the temporary file being gzip'd for RSConnect may have been cleaning up the file early on windows..)

https://github.com/rstudio/pins-python/releases/tag/v0.6.0

Please don't hesitate to re-open if you hit this issue still!
